next/src/field/schema.ts

```diff
@@ -1,18 +1,76 @@
-import type { JsfSchema } from '../types'
+import type { JsfObjectSchema, JsfSchema, NonBooleanJsfSchema } from '../types'
 import type { Field } from './type'
-import { buildFieldsObject } from './object'
+import { buildFieldObject } from './object'
 
-export function buildFieldsSchema(params: {
-  schema: JsfSchema
-}): Field[] {
-  const { schema } = params
+/**
+ * Get the input type for a field
+ * @param schema - The schema of the field
+ * @returns The input type for the field, based schema type. Default to 'text'
+ */
+function getInputType(schema: NonBooleanJsfSchema): string {
+  if (schema.type === 'string') {
+    return 'text'
+  }
 
-  if (typeof schema === 'boolean')
-    return []
+  return 'text'
+}
+
+/**
+ * Get the JSON type for a field
+ * @param schema - The non boolean schema of the field
+ * @returns The JSON type for the field, based schema type. Default to 'text'
+ */
+function getJsonType(schema: NonBooleanJsfSchema): string {
+  if (Array.isArray(schema.type)) {
+    return 'select'
+  }
+
+  if (schema.type !== undefined) {
+    return schema.type as string
+  }
+
+  return 'text'
+}
+
+/**
+ * Build a field from any schema
+ * @param schema - The schema of the field
+ * @param name - The name of the field, used if the schema has no title
+ * @param required - Whether the field is required
+ * @returns The field
+ */
+export function buildFieldSchema(
+  schema: JsfSchema,
+  name: string,
+  required: boolean = false,
+): Field | null {
+  if (typeof schema === 'boolean') {
+    return null
+  }
 
   if (schema.type === 'object') {
-    return buildFieldsObject({ schema })
+    const objectSchema: JsfObjectSchema = { ...schema, type: 'object' }
+    return buildFieldObject(objectSchema, name, required)
+  }
+
+  if (Array.isArray(schema.type)) {
+    throw new TypeError('Array type is not yet supported')
+  }
+
+  const inputType = getInputType(schema)
+
+  const field: Field = {
+    ...schema['x-jsf-presentation'],
```

Comment:
One thing I did here that I'm not enterely sure if it's correct is spread `x-jsf-presentation` to field before setting `inputType`, `type`, `name`, and so on. This means `x-jsf-presentation` attributes can't override those keys. This is however possible in v0.

Reply:
I agree with it. I don't remember any case where we would need to override those. And if we really need, then we'll catch it and fix it.